packages/hoppscotch-app/newstore/MQTTSession.ts

```diff
@@ -0,0 +1,190 @@
+import { pluck, distinctUntilChanged } from "rxjs/operators"
+import { Client as MQTTClient } from "paho-mqtt"
+import DispatchingStore, { defineDispatchers } from "./DispatchingStore"
+import {
+  HoppRealtimeLog,
+  HoppRealtimeLogLine,
+} from "~/helpers/types/HoppRealtimeLog"
+
+type HoppMQTTRequest = {
+  endpoint: string
+}
+
+type HoppMQTTSession = {
+  request: HoppMQTTRequest
+  connectingState: boolean
+  connectionState: boolean
+  subscriptionState: boolean
```

Comment:
Store session objects implement should only have data which needs to be persisted over time.

We do not need to Connecting/Connection States on the store because we don't need to save it to the disk.

Just having logs and request data is enough.